appservice/src/getKuduClient.ts

```diff
@@ -7,22 +7,18 @@ import { AppServicePlan, User } from 'azure-arm-website/lib/models';
 import { BasicAuthenticationCredentials } from 'ms-rest';
 import { addExtensionUserAgent } from 'vscode-azureextensionui';
 import KuduClient from 'vscode-azurekudu';
-import { ArgumentError } from './errors';
 import { localize } from './localize';
 import { SiteClient } from './SiteClient';
+import { nonNullProp } from './utils/nonNull';
 
 export async function getKuduClient(client: SiteClient): Promise<KuduClient> {
     if (!client.kuduHostName) {
-        const asp: AppServicePlan = await client.getAppServicePlan();
-        const notSupportedLinux: string = localize('notSupportedLinux', 'This operation is not supported by App Service plans with kind "{0}" and sku tier "{1}".', client.kind, asp.sku.tier);
-        throw new Error (notSupportedLinux);
+        const asp: AppServicePlan | undefined = await client.getAppServicePlan();
+        const notSupportedLinux: string = localize('notSupportedLinux', 'This operation is not supported by App Service plans with kind "{0}" and sku tier "{1}".', client.kind, asp && asp.sku && asp.sku.tier);
+        throw new Error(notSupportedLinux);
     }
     const user: User = await client.getWebAppPublishCredential();
-    if (!user.publishingUserName || !user.publishingPassword) {
-        throw new ArgumentError(user);
```

Comment:
Should we change nonNullProp to throw ArgumentError?  Does anybody depend on that?